SensusService/SensusServiceHelper.cs

```diff
@@ -520,17 +522,23 @@ public float GpsDeferralTimeMinutes
 
         #endregion
 
-        protected SensusServiceHelper()
+        protected SensusServiceHelper(IConcurrent mainSync)
         {
             if (SINGLETON != null)
+            {
                 throw new SensusException("Attempted to construct new service helper when singleton already existed.");
+            }
+
+            MainThreadSynchronizer = mainSync;
 
-            _registeredProtocols = new ObservableCollection<Protocol>();
-            _runningProtocolIds = new List<string>();
+            _registeredProtocols = new ConcurrentObservableCollection<Protocol>(new LockConcurrent());
+            _scriptsToRun        = new ConcurrentObservableCollection<Script>(MainThreadSynchronizer);
+
+            _runningProtocolIds   = new List<string>();
             _healthTestCallbackId = null;
-            _idCallback = new Dictionary<string, ScheduledCallback>();
-            _hasher = new SHA256Managed();
-            _pointsOfInterest = new List<PointOfInterest>();
+            _idCallback           = new Dictionary<string, ScheduledCallback>();
+            _hasher               = new SHA256Managed();
+            _pointsOfInterest     = new List<PointOfInterest>();
```

Comment:
You don't have to maintain it. 

I've got a tool that allows me to do it in one key press. Would you still like me to stop?